vaadin-accordion-flow-parent/vaadin-accordion-flow/src/main/java/com/vaadin/flow/component/accordion/AccordionPanel.java

```diff
@@ -119,4 +133,64 @@ public AccordionPanel(Component summary, Component... components) {
         this(summary);
         addContent(components);
     }
+
+    /**
+     * Sets the summary container component.
+     */
+    protected void setSummaryContainer() {
+        summaryContainer = new AccordionHeading();
+        summaryContainer.getElement().setAttribute("slot", "summary");
+        getElement().appendChild(summaryContainer.getElement());
+    }
+
+    /**
+     * Sets the component summary
+     *
+     * @see #getSummary()
+     * @param summary
+     *            the summary component to set, or <code>null</code> to remove
+     *            any previously set summary
+     */
+    @Override
+    public void setSummary(Component summary) {
```

Comment:
Instead of duplicating all methods for the summary, we could refactor `setSummaryContainer` into something like `createSummaryContainer`, which would only create and return the container, and then have the `Details` constructor do the common stuff (set slot, append element, set `summaryContainer` field).